src/prefect/projects/base.py

```diff
@@ -230,6 +233,7 @@ def initialize_project(name: str = None, recipe: str = None) -> List[str]:
     elif recipe is None:
         recipe = "local"
 
+    formatting_kwargs.update(inputs or {})
```

Comment:
probably a dumb python q - why not set `inputs: dict = {}` as a default arg?

Reply:
@parkedwards not a dumb question at all, it's a common python gotcha actually; because function definitions are run only once per-import, all default values are set and bound once and only once.  This means that if a default is a mutable object, it can begin to leak state between calls of the function.  Easy to inspect example:

```python
def f(item, x: list = []):
    x.append(item)
    return x
    
f(1) # returns [1]
f(2) # returns [1, 2]
```

So to get around that, the best practice is to use `None` as the default for mutable inputs, and initialize an empty version of the object once per-call (you'll often see things like `inputs = inputs or {}` at the top of functions for this reason).

Reply:
iiiinteresting, good to know. thanks chris